src/transactions/feed/SwapFeedItem.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import { HomeEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import TokenDisplay from 'src/components/TokenDisplay'
+import Touchable from 'src/components/Touchable'
+import SwapIcon from 'src/icons/SwapIcon'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import variables from 'src/styles/variables'
+import { useTokenInfo } from 'src/tokens/hooks'
+import { TokenExchange } from 'src/transactions/types'
+
+interface Props {
+  exchange: TokenExchange
+}
+
+function SwapFeedItem({ exchange }: Props) {
+  const { t } = useTranslation()
+  const incomingTokenInfo = useTokenInfo(exchange.inAmount.tokenAddress)
+  const outgoingTokenInfo = useTokenInfo(exchange.outAmount.tokenAddress)
+
+  const openTransferDetails = () => {
+    navigate(Screens.TransactionDetailsScreen, { transaction: exchange })
+    ValoraAnalytics.track(HomeEvents.transaction_feed_item_select)
+  }
+
+  return (
+    <Touchable disabled={false} onPress={openTransferDetails}>
+      <View style={styles.container}>
+        <View style={styles.iconContainer}>
+          <SwapIcon />
+        </View>
+        <View style={styles.contentContainer}>
+          <Text style={styles.title} testID={'SwapFeedItem/title'}>
+            {t('swapScreen.title')}
+          </Text>
+          <Text style={styles.subtitle} testID={'SwapFeedItem/subtitle'}>
+            {outgoingTokenInfo?.symbol} {'to'} {incomingTokenInfo?.symbol}
+          </Text>
+        </View>
+        <View style={styles.contentContainer}>
```

Comment:
i would recommend to remove this style, so that this box and the description box are not always the same width.

i think you can then have a style on this box `alignItems: flex-end` and remove the right align styles on the TokenDisplay components (i.e. the marginLeft and textAlign).

Reply:
I tried removing it as detailed above and when I used large fonts I noticed the display of the swap gets distorted.
#### Before and After the recommended change
<img src="https://user-images.githubusercontent.com/26950305/184440462-8db6d0f8-c7b6-4281-8459-70c508842ef0.png" width="48%" height="auto" />
<img src="https://user-images.githubusercontent.com/26950305/184440658-091a4ceb-f7e6-43da-9b22-288467adc652.png" width="48%" height="auto" />

Reply:
I added `tokenAmountContainer` and did what @kathaypacific recommended.
I also made it not to be screwed up with the extra large-size fonts!